src/vm/eventpipeconfiguration.cpp

```diff
@@ -415,16 +415,12 @@ void EventPipeConfiguration::DeleteDeferredProviders()
         EventPipeProvider *pProvider = pElem->GetValue();
         if(pProvider->GetDeleteDeferred())
         {
-            // The act of deleting the provider unregisters it and removes it from the list.
+            // The act of deleting the provider unregisters it,
+            // removes it from the list, and deletes the list element
```

Comment:
If deleting the provider deletes the list element, pElem becomes invalid. Thus, you should not be referencing it a few lines below to get the next element.

Reply:
Thanks for catching that